eng/scripts/Invoke-GeneratAndBuildV2.ps1

```diff
@@ -8,33 +8,78 @@ param (
 
 $inputJson = Get-Content $inputJsonFile | Out-String | ConvertFrom-Json
 $swaggerDir = $inputJson.specFolder
+$swaggerDir = Resolve-Path $swaggerDir
 $swaggerDir = $swaggerDir -replace "\\", "/"
+[string[]] $inputFilePaths = $inputJson.changedFiles;
 $readmeFiles = $inputJson.relatedReadmeMdFiles
 $commitid = $inputJson.headSha
 $repoHttpsUrl = $inputJson.repoHttpsUrl
+$downloadUrlPrefix = $inputJson.installInstructionInput.downloadUrlPrefix
+[string] $autorestConfig = $inputJson.autorestConfig
 
-$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
+$autorestConfigYaml = ""
+if ( $null -ne $autorestConfig -and $autorestConfig -ne "") {
+    $autorestConfig | Set-Content "config.md"
+    $autorestConfigYaml = Get-Content -Path .\config.md
+    $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+    if ( $range.count -gt 1) {
+        $startNum = $range[0];
+        $lines = $range[1] - $range[0] - 1
+        $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
+    }
+    Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+    Import-Module powershell-yaml
+    $yml = ConvertFrom-YAML $autorestConfigYaml
+    $requires = $yml["require"]
+    if ($requires.Count -gt 0) {
+        $readmeFiles = $requires
+    }
+}
 
-foreach ( $relateReadmeFile in $readmeFiles ) {
-    $readmeFile = $relateReadmeFile -replace "\\", "/"
-    
-    $sdkPath =  (Join-Path $PSScriptRoot .. ..)
-    $sdkPath = Resolve-Path $sdkPath
-    $sdkPath = $sdkPath -replace "\\", "/"
+$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
 
-    
+$sdkPath =  (Join-Path $PSScriptRoot .. ..)
+$sdkPath = Resolve-Path $sdkPath
+$sdkPath = $sdkPath -replace "\\", "/"
+for ($i = 0; $i -le $readmeFiles.Count - 1; $i++) {
+    $readmeFile = $readmeFiles[$i] -replace "\\", "/"
     $readme = ""
-    if ($commitid -ne "") {
-    if ($repoHttpsUrl -ne "") {
-        $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+    if ( $swaggerDir) {
+        $readme = (Join-Path $swaggerDir $readmeFile)
+    } elseif ( $commitid) {
+        if ($repoHttpsUrl) {
+            $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+        } else {
+            $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
+        }
     } else {
-        $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
+        Write-Error "No readme File path provided."
+        exit 1
```

Comment:
Tip: you can do both with just `throw "No readme file path provided."`. It will err and exit with err all at the same time.